lib/Backend/GlobOpt.cpp

```diff
@@ -3825,32 +3826,157 @@ GlobOpt::TestAnyArgumentsSym()
+            Js::ArgSlot formalsCount = instr->m_func->GetJnFunction()->GetInParamsCount() - 1;
+
+            Func * func = instr->m_func;
+
+            IR::Instr * argOutInstr = instr->GetSrc2()->GetStackSym()->GetInstrDef();
+
+            //The argout immediately before the InlineeStart will be the ArgOut for NewScObject
+            //So we don't want to track the stack sym for this argout.- Skipping it here.
+            if (instr->m_func->IsInlinedConstructor())
+            {
+                Assert(argOutInstr->GetSrc1()->GetStackSym()->GetInstrDef() != nullptr);
+                Assert(argOutInstr->GetSrc1()->GetStackSym()->GetInstrDef()->m_opcode == Js::OpCode::NewScObjectNoCtor);
+                argOutInstr = argOutInstr->GetSrc2()->GetStackSym()->GetInstrDef();
+            }
+            if (formalsCount < actualsCount)
+            {
+                Js::ArgSlot extraActuals = actualsCount - formalsCount;
+
+                //Skipping extra actuals passed
+                for (Js::ArgSlot i = 0; i < extraActuals; i++)
+                {
+                    argOutInstr = argOutInstr->GetSrc2()->GetStackSym()->GetInstrDef();
+                }
+            }
+
+            StackSym * undefinedSym = nullptr;
+
+            for (Js::ArgSlot param = formalsCount; param > 0; param--)
+            {
+                StackSym * argOutSym = nullptr;
+
+                if (argOutInstr->GetSrc1())
+                {
+                    if (argOutInstr->GetSrc1()->IsRegOpnd())
+                    {
+                        argOutSym = argOutInstr->GetSrc1()->GetStackSym();
+                    }
+                    else
+                    {
+                        //TODO: saravind: Can I just use the same source operand like this?
+                        // We will always have ArgOut instr - so the source operand will not be removed.
+                        argOutSym = StackSym::New(func);
+                        IR::Opnd * srcOpnd = argOutInstr->GetSrc1();
+                        IR::Opnd * dstOpnd = IR::RegOpnd::New(argOutSym, TyVar, func);
+                        IR::Instr * assignInstr = IR::Instr::New(Js::OpCode::Ld_A, dstOpnd, srcOpnd, func);
+                        instr->InsertBefore(assignInstr);
+                    }
+                }
+
+                Assert(!func->HasStackSymForFormal(param - 1));
+
+                if (param <= actualsCount)
+                {
+                    Assert(argOutSym);
+                    func->TrackStackSymForFormalIndex(param - 1, argOutSym);
+                    argOutInstr = argOutInstr->GetSrc2()->GetStackSym()->GetInstrDef();
+                }
+                else
+                {
+                    /*When param is out of range of actuals count, load undefined*/
+                    // TODO: saravind: This will insert undefined for each of the param not having an actual. - Clean up this by having a sym for undefined on func ?
```

Comment:
done.

---
In reply to: [67736799](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r67736799) [](ancestors = 67736799)